### ISSUE TITLE
DateTime serialization

### ISSUE THREAD

**Issue by promontis** [ch-robinson/dotnet-avro#39]
Hi @dstelljes!

Is it possible to create a schema with `"logicalType": "date"` instead of `string`?

**Comment by dstelljes** [comment]
It’s possible to serialize/deserialize with the timestamp logical types, but we don’t currently have a CLI flag or support on the schema builder. Would definitely be open to adding that, though.

**Comment by promontis** [comment]
So if I change `TimestampSchemaBuilderCase`  to use 
```
 var schema = new LongSchema
            {
                LogicalType = new MillisecondTimestampLogicalType()
            };
```
would that be enough?

**Comment by dstelljes** [comment]
You shouldn’t have to make any changes to the resolver—it‘ll already return a `TimestampResolution` for `DateTime`/`DateTimeOffset` (and a `DurationResolution` for `TimeSpan`).

Because ms/μs are both options, we probably want something like this instead of a boolean:

```csharp
enum TemporalBehavior
{
    Iso8601,
    EpochMicroseconds,
    EpochMilliseconds
}
```

That could be specified as a parameter on the schema builder ([the constructor without a case builders parameter](https://github.com/ch-robinson/dotnet-avro/blob/master/src/Chr.Avro/Abstract/SchemaBuilder.cs#L91)).

**Comment by promontis** [comment]
Yeah, it does return a `TimestampResolution`. The `SchemaBuilder` returns a `StringSchema` instead of a `LongSchema`. So ok if I implement the switch there? https://github.com/ch-robinson/dotnet-avro/blob/master/src/Chr.Avro/Abstract/SchemaBuilder.cs#L792

**Comment by dstelljes** [comment]
Yep! We do want to keep `"string"` as the default behavior, so add a constructor on `TimestampSchemaBuilderCase` that takes the customization parameter and then add logic to return the correct schema.

**Comment by promontis** [comment]
@dstelljes like this? #40

**Comment by dstelljes** [comment]
Released with 3.1.0. Thanks again!
